src/sparsezoo/analyze_v2/parameter_analysis.py

```diff
@@ -0,0 +1,208 @@
+        data = get_parameter_counts(self.model_graph, self.node)
+        return {
+            grouping: SparsityAnalysisModel(**counts_dict).dict()
+            for grouping, counts_dict in data.items()
+        }
+
+    def get_bits(self) -> Dict[str, Union[int, float]]:
+        """Get the number of bits and quantized bits from weights"""
+        data = get_parameter_bits(self.model_graph, self.node)
+        return {
+            grouping: QuantizationAnalysisModel(**bits_dict).dict()
+            for grouping, bits_dict in data.items()
+        }
+
+    def get_distribution(self) -> Dict[str, Union[int, float]]:
+        """Get statistics"""
+        distribution_dct = get_parameter_distribution(self.model_graph, self.node)
+        return DistributionAnalysisModel(**distribution_dct).dict()
+
+    def to_dict(self) -> Dict[str, Any]:
+        return dict(
+            name=self.node.name,
+            op_type=self.node.op_type,
+            distribution=self.distribution,
+            sparsity=self.counts,
+            quantization=self.bits,
+        )
+
+    def to_yaml(self) -> str:
+        return yaml.dump(self.to_dict())
+
+
+def get_parameter_counts(
+    model_graph: ONNXGraph,
+    node: NodeProto,
+) -> Dict[str, Union[int, float]]:
+    """Get the number of parameters in the node, if any"""
+    num_sparse_weights_four_blocks, num_weights_four_block = 0, 0
+    num_weights, _, num_weights_sparse = get_node_param_counts(node, model_graph)
+
+    if num_weights > 0:
+        (
+            num_sparse_weights_four_blocks,
+            num_weights_four_block,
+        ) = get_node_num_four_block_zeros_and_size(model_graph, node)
+
+    return {
+        "single": {
+            "counts": num_weights,
+            "counts_sparse": num_weights_sparse,
+            "percent": num_weights_sparse / num_weights if num_weights > 0 else 0,
+        },
+        "block4": {
+            "counts": num_weights_four_block,
+            "counts_sparse": num_sparse_weights_four_blocks,
+            "percent": num_sparse_weights_four_blocks / num_weights_four_block
+            if num_weights > 0
+            else 9,
+        },
+    }
+
+
+def get_parameter_bits(
+    model_graph: ONNXGraph,
+    node: NodeProto,
+    *args,
+    **kwargs,
+) -> Dict[str, Union[int, float]]:
+    """
+    Get the number of bits used to store the array
+    If the layer is quantized, assume all its elements in the ndarray
+     are quantized
+    """
+    bits = 0
+    node_weight = get_node_weight(model_graph, node)
+    if node_weight is not None and node_weight.size > 0:
+        bits = get_node_weight_bits(model_graph, node)
+
+    bits_quant = bits * is_quantized_layer(model_graph, node)
+    return {
+        "tensor": {
+            "bits": bits,
+            "bits_quant": bits * is_quantized_layer(model_graph, node),
+            "percent": bits_quant / bits if bits > 0 else 0,
+        },
+        # TODO: Channels
```

Comment:
Intended?

Reply:
Yes, because not needed for sparsezoo ui for now. We'll need it for sparsify.